front/migrations/20250626_move_process_actions_to_mcp.ts

```diff
@@ -0,0 +1,421 @@
+import type { Logger } from "pino";
+import type { CreationAttributes } from "sequelize";
+import { Op } from "sequelize";
+
+import { DEFAULT_CONVERSATION_EXTRACT_ACTION_NAME } from "@app/lib/actions/constants";
+import type { ActionBaseParams } from "@app/lib/actions/mcp";
+import type {
+  ExtractQueryResourceType,
+  ExtractResultResourceType,
+} from "@app/lib/actions/mcp_internal_actions/output_schemas";
+import type { ProcessActionOutputsType } from "@app/lib/actions/process";
+import { getWorkspaceInfos } from "@app/lib/api/workspace";
+import { Authenticator } from "@app/lib/auth";
+import {
+  AgentMCPAction,
+  AgentMCPActionOutputItem,
+  AgentMCPServerConfiguration,
+} from "@app/lib/models/assistant/actions/mcp";
+import { AgentProcessAction } from "@app/lib/models/assistant/actions/process";
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
+import { AgentMessage } from "@app/lib/models/assistant/conversation";
+import { MCPServerViewResource } from "@app/lib/resources/mcp_server_view_resource";
+import { concurrentExecutor } from "@app/lib/utils/async_utils";
+import { makeScript } from "@app/scripts/helpers";
+import { runOnAllWorkspaces } from "@app/scripts/workspace_helpers";
+import type { LightWorkspaceType, ModelId, TimeFrame } from "@app/types";
+import { isGlobalAgentId } from "@app/types";
+
+const WORKSPACE_CONCURRENCY = 50;
+const BATCH_SIZE = 200;
+const CREATION_CONCURRENCY = 50;
+
+const NOT_FOUND_MCP_SERVER_CONFIGURATION_ID = "unknown";
+
+function getTimeFrameUnit(processAction: AgentProcessAction): TimeFrame | null {
+  if (
+    processAction.relativeTimeFrameUnit &&
+    processAction.relativeTimeFrameDuration
+  ) {
+    return {
+      duration: processAction.relativeTimeFrameDuration,
+      unit: processAction.relativeTimeFrameUnit,
+    };
+  }
+
+  return null;
+}
+
+function agentProcessActionToAgentMCPAction(
+  processAction: AgentProcessAction,
+  agentConfiguration: AgentConfiguration | null,
+  mcpServerViewForExtractId: ModelId,
+  logger: Logger
+): {
+  action: ActionBaseParams & CreationAttributes<AgentMCPAction>;
+} {
+  logger.info(
+    {
+      mcpServerViewForExtractId,
+    },
+    "Found MCP server view ID for extract_data"
+  );
+
+  const extractMcpServerConfiguration =
+    agentConfiguration?.mcpServerConfigurations.find(
+      (config) => config.mcpServerViewId === mcpServerViewForExtractId
+    );
+
+  const isJITServerAction =
+    processAction.functionCallName === DEFAULT_CONVERSATION_EXTRACT_ACTION_NAME;
+
+  // Determine the MCP server configuration ID to use.
+  let mcpServerConfigurationId: string;
+
+  if (isJITServerAction) {
+    // For JIT server actions (default extract), use the hardcoded -1 ID like in the MCP
+    // server implementation.
+    mcpServerConfigurationId = "-1";
+  } else {
+    // For custom agent configurations, use the extract configuration.
+    mcpServerConfigurationId =
+      extractMcpServerConfiguration?.sId ??
```

Comment:
Actually we decided to use `.id` here. Retrieval was run before, but we will apply proper foreign key logic with @aubin-tchoi.

```suggestion
      extractMcpServerConfiguration?.id ??
```